src/main/webapp/js/instructorFeedbackResults.js

```diff
@@ -209,15 +209,43 @@ function toggleCollapse(e, pans) {
     var expand = 'Expand';
     var collapse = 'Collapse';
     var panels = pans || $('div.panel-collapse');
+    var shouldExpandPanels = $(e).html().trim().startsWith(expand);
+
+    if (shouldExpandPanels) {
+        // Expand each panel if not yet expanded.
+        // When a panel has class ajax_auto or ajax-response-auto, the panel data has not been loaded yet.
+        // We will load the data by Ajax, and ajax_auto or ajax-response-auto class will be removed.
+        // {@link #toggleCollapse} binds with clicking on both collapse panels button(button for all panels)
+        // and collapse panels button section (button for one panel). Class ajax_auto or ajax-response-auto
+        // is found in {@code panels} only if {@link #toggleCollapse} is invoked by clicking on collapse
+        // panels button.
+        var mightNeedAjaxLoading = $(e).is($('#collapse-panels-button'));
```

Comment:
In this case (and in the case of `shouldExpandPanels` above) you have a situation like this:
* The code says: `var mustShoot = wifeDrinkingTea;` 
* The explanation: because wife is allergic to tea and if she is drinking tea she must not be the real wife but an alien disguised as wife so therefore I must shoot her before she kills me first. 

I'm exaggerating, but you see my problem right? The variable assignment statement does not tell the whole story and I'm not sure the reader is able to figure out the missing links.  You might need more closer-to-truth variable names and/or explanatory comments. The explanation you gave above (for both cases) gives me a rough idea but I'm not entirely clear either. It's hard to explain this sort of things -- but must keep trying until we get it right.